cirq-core/cirq/experiments/parallel_readout_calibration.py

```diff
@@ -0,0 +1,136 @@
+from cirq.experiments.readout_experiment_result import ReadoutExperimentResult
+
+if TYPE_CHECKING:
+    import cirq
+
+
+def estimate_parallel_readout_errors(
+    sampler: 'cirq.Sampler',
+    *,
+    qubits: Iterable['cirq.Qid'],
+    trials: int = 20,
+    repetitions: int = 1000,
+    trials_per_batch: Optional[int] = None,
+    bit_strings: Optional[List[int]] = None,
+) -> ReadoutExperimentResult:
+
+    """Estimate readout error for qubits simultaneously.
+
+    For each trial, prepare a bitstring of random |0〉and |1〉states for
+    each state.  Measure each qubit.  Capture the errors per qubit of
+    zero and one state over each triel.
+
+    Args:
+        sampler: The quantum engine or simulator to run the circuits.
+        qubits: The qubits being tested.
+        repetitions: The number of measurement repetitions to perform for
+            each trial.
+        trials: The number of bitstrings to prepare.
+        trials_per_batch:  If provided, split the experiment into batches
+            with this number of trials in each batch.
+        bit_strings: A list of ints that specifies the bit strings for each
+            qubit.  There should be exactly one int per qubit.  Bits
+            represent whether the qubit is |0〉or |1〉when initialized.
+            Trials are done from least significant bit to most significant bit.
+            If not provided, the function will generate random bit strings
+            for you.
+
+    Returns:
+        A ReadoutExperimentResult storing the readout error
+        probabilities as well as the number of repetitions used to estimate
+        the probabilities. Also stores a timestamp indicating the time when
+        data was finished being collected from the sampler.  Note that,
+        if there did not exist a trial where a given qubit was set to |0〉,
+        the zero-state error will be set to `nan` (not a number).  Likewise
+        for qubits with no |1〉trial and one-state error.
+    """
+    qubits = list(qubits)
+
+    if trials <= 0:
+        raise ValueError("Must provide non-zero trials for readout calibration.")
+    if repetitions <= 0:
+        raise ValueError("Must provide non-zero repetition for readout calibration.")
+    if bit_strings is None:
+        bit_strings = [random.getrandbits(trials) for _ in qubits]
+    if len(bit_strings) != len(qubits):
+        raise ValueError(
+            f'If providing bit_strings, # of bit strings ({len(bit_strings)}) '
+            f'must equal # of qubits ({len(qubits)})'
+        )
+
+    all_circuits = []
+    all_sweeps: List[study.Sweepable] = []
+    if trials_per_batch is not None:
+        if trials_per_batch <= 0:
+            raise ValueError("Must provide non-zero trials_per_batch for readout calibration.")
+        num_batchs = trials // trials_per_batch
+        if trials % trials_per_batch > 0:
+            num_batchs += 1
+    else:
+        num_batchs = 1
+        trials_per_batch = trials
```

Comment:
What do we think of this slight reworking ? It's a little easier to digest I think.

```python3
if trials_per_batch is None:
   trials_per_batch = trials

if trials_per_batch <= 0:
  raise ValueError("Must provide non-zero trials_per_batch for readout calibration.")

num_batches = (trials + trials_per_batch - 1) // trials_per_batch

```

Reply:
Done.